collectors/google_pubsub/pubsub_collector.go

```diff
@@ -2,24 +2,48 @@ package google_pubsub
 
 import (
 	"cloud.google.com/go/pubsub"
-
+	"context"
+	"fmt"
 	"github.com/pfortin-urbn/stalk/collectors"
+	"sync"
 )
 
 type PubSubCollector struct {
 	*collectors.BaseCollector
+	PubSubClient *pubsub.Client
+	Context      context.Context
```

Comment:
context variables should not be kept here since they change for every request and you should only have to instantiate the collector once.  In go it is perfectly ok to pass the context as a parameter to the functions you are calling.

Reply:
Updated - had the functions create their own context for now to handle the interfacing.

Reply:
I tried this change, and it seems like the context needs to be tied to the pubsub client otherwise we can't cancel properly on the second go round (not entirely sure why?).  Seems like either we swarm and create a new client for each poll, and then let it die with the GetMessages scope, or we can singleton like we are here, but we need to change the interface and pass the context around like you suggested.  I think the library designed the client to be reused so maybe number 2 is the right answer?  Thoughts?

Reply:
Ok, ignore ^, you were right we can do request specific context.  I have a change coming in that should fix this.